The PostCommit Java IO Performance Tests job is flaky

The PostCommit Java IO Performance Tests is failing over 50% of the time 
Please visit https://github.com/apache/beam/actions/workflows/beam_PostCommit_Java_IO_Performance_Tests.yml?query=is%3Afailure+branch%3Amaster to see the logs.

# Situation

`:it:google-cloud-platform:GCSPerformanceTest` repeatedly fails with error:

<details>

<summary>com.google.auth.oauth2.GoogleAuthException at FileBasedIOLT.java</summary>

```
org.apache.beam.it.gcp.storage.FileBasedIOLT > testTextIOWriteThenRead FAILED	
    java.lang.IllegalArgumentException at FileBasedIOLT.java:191	
        Caused by: java.lang.RuntimeException at FileBasedIOLT.java:191	
            Caused by: java.io.IOException at FileBasedIOLT.java:191	
                Caused by: com.google.auth.oauth2.GoogleAuthException at FileBasedIOLT.java:191	
                    Caused by: com.google.api.client.http.HttpResponseException at FileBasedIOLT.java:191	
org.apache.beam.it.gcp.storage.FileBasedIOLT > classMethod FAILED	
    java.lang.RuntimeException at FileBasedIOLT.java:128	
        Caused by: com.google.cloud.storage.StorageException at FileBasedIOLT.java:128	
            Caused by: com.google.auth.oauth2.GoogleAuthException at FileBasedIOLT.java:128	
                Caused by: com.google.api.client.http.HttpResponseException at FileBasedIOLT.java:128
```

</details>

# Background

History of failures go back as far as 9 months, most recently related to an error in GCP credentials. Notably, [beam_PostCommit_Java_IO_Performance_Tests.yml has an authentication step using static credentials](https://github.com/apache/beam/blob/e1245d9da136805ab5d40710e968b472b215b6c4/.github/workflows/beam_PostCommit_Java_IO_Performance_Tests.yml#L91-L95). Searching through the repository we see this `google-github-actions/auth@v1` pattern [used in only a few places](https://github.com/search?q=repo%3Aapache%2Fbeam%20google-github-actions%2Fauth%40v1&type=code) while there are numerous tests and workflows requiring GCP authentication.
- 

# Assessment

The test may be improperly configured using incorrect credentials. More importantly, it uses an outdated pattern via static credentials to authenticate to Google cloud.

# Recomendation

Remove the use of `google-github-actions/auth@v1`.

--- a/.github/trigger_files/beam_PostCommit_Java_IO_Performance_Tests.json
+++ b/.github/trigger_files/beam_PostCommit_Java_IO_Performance_Tests.json
@@ -1,0 +1,4 @@
+{
+  "comment": "Modify this file in a trivial way to cause this test suite to run",
+  "modification": 2
+}

--- a/.github/workflows/beam_PostCommit_Java_IO_Performance_Tests.yml
+++ b/.github/workflows/beam_PostCommit_Java_IO_Performance_Tests.yml
@@ -88,11 +88,6 @@ jobs:
       uses: ./.github/actions/setup-environment-action
       with:
         java-version: default
-    - name: Authenticate on GCP
-      uses: google-github-actions/auth@v1
-      with:
-        credentials_json: ${{ secrets.GCP_SA_KEY }}
-        project_id: ${{ secrets.GCP_PROJECT_ID }}
     - name: run scheduled javaPostcommitIOPerformanceTests script
       if: github.event_name == 'schedule' #This ensures only scheduled runs publish metrics publicly by changing which exportTable is configured
       uses: ./.github/actions/gradle-command-self-hosted-action

